### PR TITLE
Send back the logged in LDAP username to haproxy

### PR DESCRIPTION
**Pull request by manu-ns**
In case of successful login using the LDAP backend, the agent now sends an SPOE message containing the username of the logged in user.

This commit also:
 - Updates the haproxy test config to write a X-Authorized-User header with the logged in username
 - Updates the nginx backend config to copy that header in a response header, to be able to test it
 - Updates TestShouldAuthenticateSuccessfullyInLDAP to test the new behavior

### CODE REVIEW

**Review by mougams — CHANGES_REQUESTED** [review]
Do you really need this merge branch review?
I'm talking about the commit named
`[Merge branch 'master' into authenticated-user](https://github.com/criteo/haproxy-spoe-auth/pull/27/commits/2b0741527272c928ae370971bfd70d73b86724c5)`

File: resources/haproxy/haproxy.cfg [hunk 1]
```diff
@@ -28,7 +28,7 @@ frontend haproxynode
     acl oauth2logout path_beg /oauth2/logout
 
     acl dex_domain hdr_beg(host) -i dex.example.com
-    
+
```
Comment (mougams) [inline resources/haproxy/haproxy.cfg:31]: Unnecessary diff

**Comment by manu-ns** [comment]
> Do you really need this merge branch review? I'm talking about the commit named `[Merge branch 'master' into authenticated-user](https://github.com/criteo/haproxy-spoe-auth/pull/27/commits/2b0741527272c928ae370971bfd70d73b86724c5)`

This is there because I merged dependencies bumps and rebased them here. It'll all be squashed on merge.

File: resources/nginx/default.conf [hunk 2]
```diff
@@ -9,6 +9,9 @@ server {
     location / {
         add_header Last-Modified $date_gmt;
         add_header Cache-Control 'no-store, no-cache, must-revalidate, proxy-revalidate, max-age=0';
+        if ($http_X_Authorized_User) {
+            add_header Request-X-Authorized-User $http_X_Authorized_User;
```
Comment (mougams) [inline resources/nginx/default.conf:13]: WDYT abour renaming to X-Request-Authorized-User? (starting with X- instead)

**Review by mougams — CHANGES_REQUESTED** [review]
Minor comments

File: resources/nginx/default.conf (hunk 2 above)
Comment (manu-ns) [reply resources/nginx/default.conf:13]: This was to mirror what we do on other probes, where we prepend `Request-` to all headers, while using `X-` could mean it's an actual response header

**Review by mougams — APPROVED** [review]
